cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/CountTest.scala

```diff
@@ -0,0 +1,163 @@
+                  #(fido)-[:HAS_TOY]->(:Toy{name:'Banana'})""".stripMargin('#'))
+    synopsis("An count subquery `COUNT` can be used to find out how many times a specified pattern appears in the data.")
+    p(
+      """
+        |* <<count-subqueries, Using count subqueries in WHERE>>
+        |** <<count-subquery-simple-case, Simple count subquery>>
+        |** <<count-subquery-with-where, Count subquery with `WHERE` clause>>
+        |** <<count-subquery-nesting, Nesting count subqueries>>
+        |* <<count-subqueries-other-clauses, Using count subqueries inside other clauses>>
+        |** <<count-subqueries-with-return, Using count subqueries with `RETURN` clause>>
+        |** <<count-subqueries-with-set, Using count subqueries with `SET` clause>>
+        |** <<count-subqueries-with-case, Using count subqueries with `CASE` expression>>
+        |** <<count-subqueries-with-with, Using count subqueries with `WITH` expression>>
+      """.stripMargin)
+    p("""#It can be used in the same way as a path pattern but it allows you to use `WHERE` clauses internally.
+        #A subquery has a scope, as indicated by the opening and closing braces, `{` and `}`.
+        #Any variable that is defined in the outside scope can be referenced inside the subquery's own scope.
+        #Variables introduced inside the subquery are not part of the outside scope and therefore can't be accessed on the outside.""".stripMargin('#'))
+    functionWithCypherStyleFormatting("""COUNT {
+                                        #  [Pattern]
+                                        #  WHERE [Expression]
+                                        #}""".stripMargin('#'))
+    p("The `WHERE` clause is optional.")
+    p("It is worth noting that the `MATCH` keyword can be omitted in subqueries and that the `WHERE` clause is optional.")
+    section("Using count subqueries in `WHERE`", "count-subqueries") {
+      section("Simple count subquery", "count-subquery-simple-case") {
+        p("""Variables introduced by the outside scope can be used in the inner `MATCH` clause. The following query exemplifies this and would output the owners of more tahan one dog:""")
+        query("""MATCH (person:Person)
+                #WHERE COUNT { (person)-[:HAS_DOG]->(:Dog) } > 1
+                #RETURN person.name AS name""".stripMargin('#'),
+          ResultAssertions(r => {
+            r.toList should equal(List(Map("name" -> "Peter")))
+          })) {
+          resultTable()
+        }
+      }
+      section("Count subquery with `WHERE` clause", "count-subquery-with-where") {
+        p("""A `WHERE` clause can be used inside the `COUNT` pattern.
+            #Variables introduced by the `MATCH` clause and the outside scope can be used in this scope.""".stripMargin('#'))
+        query("""MATCH (person:Person)
+                #WHERE COUNT {
+                #  (person)-[:HAS_DOG]->(dog:Dog)
+                #  WHERE person.name = dog.name
+                #} > 0
```

Comment:
I was considering if it would be interesting to mention that `COUNT > 0`is semantically equivalent to  `EXISTS`, but to not confuse people we should maybe wait with that until we have improved both so they work in the same places.